pkg/templates/aws/resources/region.yaml

```diff
@@ -2,7 +2,7 @@ qualified_type_name: aws:region
 display_name: Region
 
 properties:
-  RegionName:
+  Name:
```

Comment:
The original property name was to disambiguate from the resource ID's Name field. We can't technically do `fieldValue` or `fieldRef` on those, but that may not be readily apparent and/or could be changed in the future.

Reply:
i dont think we ever need fieldValue or fieldRef on the id name because you can access it via .Name and we sub the name in properties if it ever changes.

Reply:
Only case I could see for that might be for description-type fields like in a subnet rule or environment variable - ie, freeform things. And the only benefit over typing it out is that it'd get updated if the resource renamed. A niche case, so definitely not blocking, just digging up what I recall for why this was named that way.